### PR TITLE
Create release_ping.yml

### PR DESCRIPTION
**Pull request by markbiek**
This is a first pass at a workflow to send a Slack notification when a new release is published.

See: (issue #)

### What and why? 🤔

Provide a description of what is being changed and why.
Reference issues, provide examples, diagrams, etc.
Be descriptive, give context, and avoid one-liners.

### Testing Steps ✍️
I don't think there's anything anyone can do testing-wise at this point. I just need to get the workflow out there so _I_ can start testing it 😄 

- [ ] Bullet point checklists
- [ ] Are useful for some tests

### You're it! 👑

Tag a random reviewer by adding `@Automattic/stream-builders` to the reviewers section, but mention them here for visibility as well.

@Automattic/stream-builders

### CODE REVIEW

**Review by sanmai — APPROVED** [review]
Code looks sane, not sure how to test it.

File: .github/workflows/release_ping.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
+      SLACK_WEBHOOK_TYPE: INCOMING_WEBHOOK
+    steps:
+      - name: Send notification for new release
+        if: github.event.action == 'published'
+        id: slack
+        uses: slackapi/slack-github-action@v1.24.0
+        env:
+          release_status: ${{ github.event.release.draft && ' draft ' || ' ' }}
+          release_icon: ${{ github.event.release.draft && ':pr-draft:' || ':pull-request:' }}
```
Comment (dhruvkb) [inline .github/workflows/release_ping.yml]: This only runs when the release is published, so the draft branch will not be exercised. Also the Slack emojis for `:pr-draft:` and `:pull-request:` are not quite appropriate for releases. They look like this:

Screenshot 2024-01-26 at 10 08 02 AM

Screenshot 2024-01-26 at 10 08 48 AM

File: .github/workflows/release_ping.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    name: Send Slack message
+    runs-on: ubuntu-latest
+    env:
+      release_url: ${{ github.event.release.html_url }}
+      release_name: ${{ github.event.release.name }}
+      SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
+      SLACK_WEBHOOK_TYPE: INCOMING_WEBHOOK
+    steps:
+      - name: Send notification for new release
+        if: github.event.action == 'published'
```
Comment (dhruvkb) [inline .github/workflows/release_ping.yml]: The workflow runs on published event only so this check is not necessary.

**Review by dhruvkb — COMMENTED** [review]
Overall this LGTM and it should work, but there is potential for further iteration on this.

One way to test would be to include a [`workflow_dispatch` trigger](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_dispatch) for the workflow (with some [inputs](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onworkflow_dispatchinputs) that can be used in place of `github.event.release`) and use that to send test messages and see if it works.

**Pull request merged**